Operations import is missing from generated foo_client.go

## Problem statement

Generated `client/foo_client.go` doesn't contain import for operations package.

## Swagger specification

```yaml
---
swagger: "2.0"
info:
  title: foo api
  version: "1.0.0"
paths:
  /version:
    get:
      responses:
        "200":
          description: foo version
          schema:
            type: string
      summary: foo version
      operationId: showVersion
```

## Steps to reproduce

Put the above yaml into `swagger.yaml` and run

    rm -rf ./pkg/foo_api && \
    mkdir -p ./pkg/foo_api && \
    docker run \
      --rm -it \
      --user "$(id -u):$(id -g)" \
      -e GOPATH=$HOME/go:/go \
      -v $HOME:$HOME \
      -w $(pwd) quay.io/goswagger/swagger \
      generate client -f ./swagger.yaml -t ./pkg/foo_api

Observe go failing to build the package:

```
$ go build ./pkg/foo_api/client/foo_client.go 
go: finding github.com/go-openapi/strfmt v0.19.5
go: finding github.com/go-openapi/runtime v0.19.15
go: finding github.com/go-openapi/runtime/client latest
go: downloading github.com/go-openapi/runtime v0.19.15
go: extracting github.com/go-openapi/runtime v0.19.15
go: downloading github.com/go-openapi/strfmt v0.19.5
go: extracting github.com/go-openapi/strfmt v0.19.5
# command-line-arguments
pkg/foo_api/client/foo_client.go:101:13: undefined: operations
```

## Environment
swagger version: v0.23.0 commit: 3596b40bedd4f06cc918767098568456d2ec2370
go version: go1.12.7 linux/amd64
OS: Ubuntu 16.04.5

--- a/fixtures/bugs/2278/fixture-2278.yaml
+++ b/fixtures/bugs/2278/fixture-2278.yaml
@@ -1,0 +1,25 @@
+---
+swagger: "2.0"
+info:
+  title: foo api
+  version: "1.0.0"
+paths:
+  /version:
+    get:
+      responses:
+        "200":
+          description: foo version
+          schema:
+            type: string
+      summary: foo version
+      operationId: showVersion
+  /withtags:
+    get:
+      tags: [ "sometag" ]
+      responses:
+        "200":
+          description: foo tagged version
+          schema:
+            type: string
+      summary: foo tagged version
+      operationId: showTaggedVersion

--- a/generator/bindata.go
+++ b/generator/bindata.go
@@ -58,7 +58,7 @@ import (
 func bindataRead(data []byte, name string) ([]byte, error) {
 	gz, err := gzip.NewReader(bytes.NewBuffer(data))
 	if err != nil {
-		return nil, fmt.Errorf("read %q: %w", name, err)
+		return nil, fmt.Errorf("read %q: %v", name, err)
 	}
 
 	var buf bytes.Buffer
@@ -66,7 +66,7 @@ func bindataRead(data []byte, name string) ([]byte, error) {
 	clErr := gz.Close()
 
 	if err != nil {
-		return nil, fmt.Errorf("read %q: %w", name, err)
+		return nil, fmt.Errorf("read %q: %v", name, err)
 	}
 	if clErr != nil {
 		return nil, err
@@ -1018,9 +1018,6 @@ var _bindata = map[string]func() (*asset, error){
 	"templates/validation/primitive.gotmpl":                       templatesValidationPrimitiveGotmpl,
 	"templates/validation/structfield.gotmpl":                     templatesValidationStructfieldGotmpl,
 }
-
-// AssetDebug is true if the assets were built with the debug flag enabled.
-const AssetDebug = false
 
 // AssetDir returns the file names below a certain
 // directory embedded in the file by go-bindata.

--- a/generator/build_test.go
+++ b/generator/build_test.go
@@ -42,6 +42,9 @@ func TestGenerateAndBuild(t *testing.T) {
 		"issue 2111": {
 			"../fixtures/bugs/2111/fixture-2111.yaml",
 		},
+		"issue 2278": {
+			"../fixtures/bugs/2278/fixture-2278.yaml",
+		},
 	}
 
 	for name, cas := range cases {

--- a/generator/client_test.go
+++ b/generator/client_test.go
@@ -266,6 +266,11 @@ func TestClient(t *testing.T) {
 				assert.True(t, fileExists(target, "operations"))
 			},
 		},
+		{
+			name:      "name with trailing API",
+			spec:      filepath.Join("..", "fixtures", "bugs", "2278", "fixture-2278.yaml"),
+			wantError: false,
+		},
 	}
 
 	for i, tt := range tests {

--- a/generator/shared.go
+++ b/generator/shared.go
@@ -755,13 +755,17 @@ func titleOrDefault(specDoc *loads.Document, name, defaultName string) string {
 }
 
 func mainNameOrDefault(specDoc *loads.Document, name, defaultName string) string {
-	// _test won't do as main server name
+	// *_test won't do as main server name
 	return strings.TrimSuffix(titleOrDefault(specDoc, name, defaultName), "Test")
 }
 
 func appNameOrDefault(specDoc *loads.Document, name, defaultName string) string {
-	// _test_api, _api_test, _test, _api won't do as app names
-	return strings.TrimSuffix(strings.TrimSuffix(strings.TrimSuffix(titleOrDefault(specDoc, name, defaultName), "Test"), "API"), "Test")
+	// *_test won't do as app names
+	name = strings.TrimSuffix(titleOrDefault(specDoc, name, defaultName), "Test")
+	if name == "" {
+		name = swag.ToGoName(defaultName)
+	}
+	return name
 }
 
 type opRef struct {

--- a/generator/shared_test.go
+++ b/generator/shared_test.go
@@ -511,6 +511,8 @@ func TestShared_AppNameOrDefault(t *testing.T) {
 	require.NotNil(t, specDoc.Spec().Info)
 	specDoc.Spec().Info.Title = "    "
 	assert.Equal(t, "Xyz", appNameOrDefault(specDoc, "  ", "xyz"))
+	specDoc.Spec().Info.Title = "test"
+	assert.Equal(t, "Xyz", appNameOrDefault(specDoc, "  ", "xyz"))
 }
 
 func TestShared_GatherModel(t *testing.T) {

